src/thing_objects.c

```diff
@@ -1694,6 +1694,38 @@ TngUpdateRet object_update_power_lightning(struct Thing *objtng)
 }
 #undef NUM_ANGLES
 
+/**
+ * Finds an empty safe adjacent position on slab.
+ * @param thing The thing which is to be moved.
+  * @param pos The target position pointer.
+ */
+TbBool find_free_position_on_slab(const struct Thing* thing, struct Coord3d* pos)
+{
+    MapSubtlCoord start_stl = ACTION_RANDOM(9);
+    int nav_sizexy = subtile_coord(thing_nav_block_sizexy(thing), 0);
+
+    for (long nround = 0; nround < 9; nround++)
```

Comment:
There is a constant for that. NUM_AROUND or something like that. Also pass it to random.